tasks/20230527-l2-gauge-checkpointer/test/task.fork.ts

```diff
@@ -0,0 +1,461 @@
+      TaskMode.READ_ONLY,
+      getForkedNetwork(hre)
+    );
+    adderCoordinator = await adderCoordinatorTask.deployedInstance('GaugeAdderMigrationCoordinator');
+
+    await authorizer.connect(daoMultisig).grantRole(await authorizer.DEFAULT_ADMIN_ROLE(), adderCoordinator.address);
+    await adderCoordinator.performNextStage();
+  });
+
+  before('get gauge relative weights and associate them with their respective address', async () => {
+    const gaugeControllerTask = new Task('20220325-gauge-controller', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    const gaugeController = await gaugeControllerTask.deployedInstance('GaugeController');
+
+    const getGaugesData = async (gaugeInputs: [string, number][]) => {
+      return Promise.all(
+        gaugeInputs.map(async (gaugeInput) => {
+          return {
+            address: gaugeInput[0],
+            weight: await gaugeController['gauge_relative_weight(address)'](gaugeInput[0]),
+            expectedCheckpoints: gaugeInput[1],
+          };
+        })
+      );
+    };
+    const singleRecipientGaugesData: GaugeData[] = await getGaugesData(singleRecipientGauges);
+    const polygonRootGaugesData: GaugeData[] = await getGaugesData(polygonRootGauges);
+    const arbitrumRootGaugesData: GaugeData[] = await getGaugesData(arbitrumRootGauges);
+    const optimismRootGaugesData: GaugeData[] = await getGaugesData(optimismRootGauges);
+    const gnosisRootGaugesData: GaugeData[] = await getGaugesData(gnosisRootGauges);
+
+    gauges.set(GaugeType.Ethereum, singleRecipientGaugesData);
+    gauges.set(GaugeType.Polygon, polygonRootGaugesData);
+    gauges.set(GaugeType.Arbitrum, arbitrumRootGaugesData);
+    gauges.set(GaugeType.Optimism, optimismRootGaugesData);
+    gauges.set(GaugeType.Gnosis, gnosisRootGaugesData);
+  });
+
+  before('check total expected checkpoints', () => {
+    let sum = 0;
+    for (const [, gaugeData] of gauges.entries()) {
+      if (gaugeData.length > 0) {
+        sum += gaugeData.map((gaugeData) => gaugeData.expectedCheckpoints).reduce((a, b) => a + b);
+      }
+    }
+    expect(sum).to.be.eq(TOTAL_EXPECTED_CHECKPOINTS);
+  });
+
+  before('add gauges to checkpointer', async () => {
+    // Some gauges were created from previous factories, so they need to be added by governance.
+    // For simplicity, we just add all of them with the same method.
+    // The non-permissioned 'addGauges' function is already tested in the unit test.
+    await authorizer
+      .connect(daoMultisig)
+      .grantRole(await actionId(L2GaugeCheckpointer, 'addGaugesWithVerifiedType'), admin.address);
+
+    await Promise.all(
+      Array.from(gauges).map(([gaugeType, gaugesData]) => {
+        L2GaugeCheckpointer.connect(admin).addGaugesWithVerifiedType(
+          GaugeType[gaugeType],
+          gaugesData.map((gaugeData) => gaugeData.address)
+        );
+      })
+    );
+  });
+
+  before('grant checkpoint permission to fees gauge checkpointer', async () => {
```

Comment:
Is this title right? "fees gauge checkpointer"?

Reply:
:grimacing: good catch.